src/main/java/org/folio/auth/authtokenmodule/AuthorizeApi.java

```diff
@@ -0,0 +1,724 @@
+        for (Object o : permissionsRequired) {
+          if (!permissions.contains(o)
+              && !extraPermissions.contains(o)) {
+            logger.error(permissions.encode() + "{} (user permissions) nor {}"
+                + " (module permissions) do not contain {}",
+                permissions.encode(), extraPermissions.encode(), o);
+            // mod-authtoken should return the module tokens header even in case of errors.
+            // If not, pre+post filters will NOT get modulePermissions from Okapi
+            ctx.response().putHeader(XOkapiHeaders.MODULE_TOKENS, moduleTokens.encode());
+            endText(ctx, 403, "Access requires permission: " + o);
+            return;
+          }
+        }
+
+        // Remove all permissions not listed in permissionsRequired or
+        // permissionsDesired
+        List<Object> deleteList = new ArrayList<>();
+        for (Object o : permissions) {
+          if (!permissionsRequired.contains(o) && !Util.arrayContainsGlob(permissionsDesired, (String) o)) {
+            deleteList.add(o);
+          }
+        }
+
+        for (Object o : deleteList) {
+          permissions.remove(o);
+        }
+
+        String finalToken;
+        try {
+          finalToken = token.encodeAsJWT(tokenCreator);
+        } catch (Exception e) {
+          String message = String.format("Error creating final token: %s", e.getMessage());
+          logger.error(message);
+          // mod-authtoken should return the module tokens header even in case of errors.
+          // If not, pre+post filters will NOT get modulePermissions from Okapi.
+          ctx.response().putHeader(XOkapiHeaders.MODULE_TOKENS, moduleTokens.encode());
+          endText(ctx, 500, "Error creating access token");
+          return;
+        }
+
+        // Return header containing relevant permissions
+        ctx.response()
+            .setChunked(true)
+            .setStatusCode(202)
+            .putHeader(MainVerticle.CONTENT_TYPE, "text/plain")
+            .putHeader(XOkapiHeaders.PERMISSIONS, permissions.encode())
+            .putHeader(XOkapiHeaders.MODULE_TOKENS, moduleTokens.encode())
+            .putHeader("Authorization", "Bearer " + finalToken)
+            .putHeader(XOkapiHeaders.TOKEN, finalToken);
+
+        if (finalUserId != null) {
+          ctx.response().putHeader(XOkapiHeaders.USER_ID, finalUserId);
+        }
+
+        ctx.response().end();
+      });
+    });
+  }
+
+  private void mergePerms(JsonArray perms, JsonArray morePerms) {
+    for (Object o : morePerms) {
+      String permName = (String) o;
+      if (!perms.contains(permName)) {
+        perms.add(permName);
+      }
+    }
+  }
+
+  public String extractToken(String authorizationHeader) {
+    // Grab anything after 'Bearer' and whitespace
+    Pattern pattern = Pattern.compile("Bearer\\s+(.+)");
+    Matcher matcher = pattern.matcher(authorizationHeader);
+    if (matcher.find() && matcher.groupCount() > 0) {
```

Comment:
```suggestion
    if (matcher.find()) {
```
`groupCount()` is always 1 when `find()` is true because `(.+)` is a required match.